gubernator/main.py

```diff
@@ -187,15 +196,17 @@ def parse_junit(xml):
             name = child.attrib['name']
             time = float(child.attrib['time'])
             for param in child.findall('failure'):
-                yield name, time, param.text
+                pod_name = get_pod_name(param.text)            
+                yield name, time, param.text, pod_name
```

Comment:
Don't do this here. The pod name can be derived from the failure text, so this parsing should occur in a different location, closer to its use.